src/terraform/providers/terraform-provider-avere/junction.go

```diff
@@ -15,11 +15,15 @@ func (j *Junction) IsEqual(j2 *Junction) bool {
 		j.CoreFilerExport == j2.CoreFilerExport &&
 		j.ExportSubdirectory == j2.ExportSubdirectory &&
 		j.PolicyName == j2.PolicyName &&
-		ExportRulesEqual(j.ExportRules, j2.ExportRules) &&
-		j.CifsShareName == j2.CifsShareName &&
+		j.CifsShareName == j2.CifsShareName
+}
+
+func (j *Junction) RequiresUpdate(j2 *Junction) bool {
```

Comment:
Just to make sure I understand correctly, this function is the heart of this update -- previously a comparison of junctions was made, and if they weren't equal it led to a complete tear down of the vFXT. Now ExportRules, CifsAces,CifsCreateMask,CifsDirMask can be updated without rebuilding the entire vFXT?

Reply:
yes, exactly, this is the core of the PR.  Previously the junction was completely removed and re-added if any part of it was changed.  This made the code easy, because we never had to add the complication (and associated bugs) of figuring out what to change, and we could just perform a delete and add.